Library/Homebrew/formula_auditor.rb

```diff
@@ -332,6 +332,13 @@ def audit_deps
       end
 
       return unless @core_tap
+
+      bad_gcc_dep = linux_only_gcc_dep?(formula) && (@strict || begin
+        fv = FormulaVersions.new(formula)
+        fv.formula_at_revision("origin/HEAD") { |prev_f| !linux_only_gcc_dep?(prev_f) }
```

Comment:
How much slowdown does this check add? I just note that `FormulaVersions` is sometimes incredibly slow.

Reply:
Will benchmark, but I don't think this is slow at all. The reason that `FormulaVersions` is slow is because of deep traversal into the Git log. We skip that by only checking `origin/HEAD`.

Reply:
Cool, thanks!

Reply:
Unfortunately, I was wrong -- while it's not slow for individual formulae (adds a couple hundredths of a second), it's slow for a tap_syntax job.

Will rework this into something that can be done when auditing a formula directly so that we can skip it for tap_syntax.

Reply:
Thanks @carlocab 👍🏻

Reply:
#13750
https://github.com/Homebrew/homebrew-test-bot/pull/821